app/models/order_detail.rb

```diff
@@ -890,6 +890,15 @@ def notify_purchaser_of_order_status
     end
   end
 
+  def self.where_order_number_and_id(query)
+    return nil unless /\A(?<order_id>\d+)-(?<id>\d+)\z/ =~ query
+    where(id: id, order_id: order_id)
```

Comment:
I wonder if we could maybe now use this in the `GlobalSearch::OrderSearcher`. It's fine for now, though.